File: scala/src/main/scala/io/digitalstream/basic/Modules.scala
/**
  * Created by sebby on 9/26/16.
  */
package io.digitalstream.basic {

  import self.deeply.nested.Utils.{foo => bar}

  object Modules {
    def run(): Unit = {
      bar()
    }
  }

  package self.deeply.nested {

    object Utils {
      def foo(): Unit = {
        println("called `deeply::nested::foo()`")
      }
    }

  }

}

File: kotlin/src/main/kotlin/io/digitalstream/basic/Modules.kt
/**
 * Created by sebby on 9/25/16.
 */
package io.digitalstream.basic

import io.digitalstream.basic.Utils.foo as bar

object Modules {

  fun run(): Unit {
    Utils.foo()
  }
}

object Utils {
  fun foo(): Unit {
    println("called `deeply::nested::foo()`")
  }
}
